src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BuldozerProducer().get());
+        }
+        if (type == Excavator.class) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        }
+        if (type == Track.class) {
+            return new ArrayList<>(new TrackProducer().get());
+        }
+        throw new ClassCastException("Unknown class [" + type + "]");
```

Comment:
ClassCastException won't be quite good here, cause it appears while casting of invalid class, for example: 
Human fred = (Human) new Object();
```suggestion
        throw new RuntimeException("Unknown class [" + type + "]");
```

Reply:
Done